src/inc/til/bitmap.h

```diff
@@ -130,36 +131,111 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
                 }
             }
         };
-    }
 
+    template<typename Allocator = std::allocator<unsigned long long>>
     class bitmap
     {
     public:
-        using const_iterator = details::_bitmap_const_iterator;
+        using allocator_type = Allocator;
+        using const_iterator = details::_bitmap_const_iterator<allocator_type>;
 
-        bitmap() noexcept :
+    private:
+        using run_allocator_type = typename std::allocator_traits<allocator_type>::template rebind_alloc<til::rectangle>;
+
+    public:
+        explicit bitmap(const allocator_type& allocator) noexcept :
+            _alloc{ allocator },
             _sz{},
             _rc{},
-            _bits{},
-            _runs{}
+            _bits{ _alloc },
+            _runs{ _alloc }
+        {
+        }
+
+        bitmap() noexcept :
+            bitmap(allocator_type{})
         {
         }
 
         bitmap(til::size sz) :
-            bitmap(sz, false)
+            bitmap(sz, false, allocator_type{})
         {
         }
 
-        bitmap(til::size sz, bool fill) :
+        bitmap(til::size sz, const allocator_type& allocator) :
+            bitmap(sz, false, allocator)
+        {
+        }
+
+        bitmap(til::size sz, bool fill, const allocator_type& allocator) :
+            _alloc{ allocator },
             _sz(sz),
             _rc(sz),
-            _bits(_sz.area()),
-            _runs{}
+            _bits(_sz.area(), fill ? std::numeric_limits<unsigned long long>::max() : 0, _alloc),

```

Comment:
functional change, but minor. instead of using set_all to fill all the bits, we are constructing the dynamic_bitset with `0xffffffffffffffff` stored in its blocks. It has the same effect!